src/process_utility.c

```diff
@@ -2390,8 +2391,13 @@ process_index_start(ProcessUtilityArgs *args)
 		CacheInvalidateRelcacheByRelid(info.obj.objectId);
 	}
 
+	CommitTransactionCommand();
+	StartTransactionCommand();
+
 	UnlockRelationIdForSession(&main_table_index_lock_relid, AccessShareLock);
 
+	DEBUG_WAITPOINT("indexing_done");
```

Comment:
Added the function name here (I think I will refactor the waitpoints to enforce the function name as well, otherwise this will be unmanageable long-term).